tiledb/sm/enums/query_status.h

```diff
@@ -44,6 +49,44 @@ enum class QueryStatus : char {
 #undef TILEDB_QUERY_STATUS_ENUM
 };
 
+/** Returns the string representation of the input querystatus type. */
+inline const std::string& query_Status_str(QueryStatus query_Status) {
+  switch (query_Status) {
+    case QueryStatus::FAILED:
+      return constants::query_status_failed_str;
+    case QueryStatus::COMPLETED:
+      return constants::query_status_completed_str;
+    case QueryStatus::INPROGRESS:
+      return constants::query_status_inprogress_str;
+    case QueryStatus::INCOMPLETE:
+      return constants::query_status_incomplete_str;
+    case QueryStatus::UNINITIALIZED:
+      return constants::query_status_uninitialized_str;
+    default:
+      assert(0);
+      return constants::empty_str;
+  }
+}
+
+/** Returns the query status given a string representation. */
+inline Status query_status_enum(
```

Comment:
Do these functions have to be defined in the header file?

Reply:
They are defined in the header file for the existing functions (datatype, query_type, compression, etc), so I defined them in the header for query_status.

Reply:
Ok, at some point we should probably go back and move them to source files in the interest of not increasing compile time more than necessary, but this is fine for now.